### PR TITLE
Remove duplicate perl modules and add CI check

### PR DESCRIPTION
**Pull request by Flamefire**
- [ ] Requires https://github.com/easybuilders/easybuild-easyblocks/pull/2386

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/040790fa85b39b2a306d8757aeb87943 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/7420a5ff8fb6697b552d8cc594ee6ca1 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/a82b2c2dee9ace1fa4d7d89e57db6e05 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/5ae26948ea013f25acaae65b6bcdd4cd for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/7f5c0a02118cc8f3343b2cc841b3b580 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/ae7ce5695492e49f5a900709118df75a for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/05d7048b9bfb963973100e3e158912de for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/d9ab29f4376e674480e5536986c921a9 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/a131819ac7df2b835a36bbc418587e9b for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2386
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in total)
taurusa4 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/f8a1a039c0b7027628328a40d9fb318c for a full test report.

**Comment by Micket** [comment]
Just to that I understand what's happening here: Do we need the alternative names for some stuff so thats why they are these redefined packages?
(I'm unfamiliar with Perls module system)

**Comment by Flamefire** [comment]
Oh, sorry for the noise, test reports with `--job` are not a good idea it seems ;)

In R we have packages only by name. Those are supposed to be contained in the R base installation. Similar in Perl there seem to be packages which are part of other packages. The only reason why those are added again appears to be to make sure they exist by checking for them in the sanity check. So the change here is to NOT install those (which leads to installing the same package exactly the same way again), but only use their name so that is sanity checked.
I'm not familiar with Perl myself, so that is what I understood.

TLDR: Allow adding sanity checks for packages which are sub-packages of already installed packages (basically)

**Comment by Micket** [comment]
Right. In some cases here, we don't actually have "sub" packages, but they are on the same level, like IO::Handle and friends under https://metacpan.org/release/IO
Perhaps it would be better to just have "IO" as the collection name instead of trying to list a specific submodule?
And, if so, just flat out don't allow duplicates be happy with that?

I don't think many are intentionally added IO::File to be sure.. it's almost certainly just accidental because someone (like me) didn't know there wasn't a 1-to-1 mapping of module to source in Perl.

Some of these also seem to be added despite them being part of Perl itself
Like all the IO stuff also show up here https://metacpan.org/release/perl so we shouldn't use any of those.

**Comment by Flamefire** [comment]
>  Perhaps it would be better to just have "IO" as the collection name instead of trying to list a specific submodule?

Similar to Python ECs we use the name to do some kind of `import <pkg>` as a sanity check if that module exists. So collections don't seem possible.

If some of them are part of Perl we could still use this PR and in a follow-up "downgrade" them to the module name only, so they are checked for but not installed, similar to the R ECs. But that got to come from someone with more experience with Perl.

This PR is good as a check for the EasyBlock change and future removals/changes can be based on this :)

**Comment by Flamefire** [comment]
@boegel @Micket There wasn't any further reply here. So what is the state?

**Comment by boegel** [comment]
This has bitrotted quite a bit...

@Flamefire Do you think it's worth opening a new PR with the additional check in the test suite, and tackling more recent Perl easyconfigs accordingly?

Some of the ones touched in here have been archived already...

**Comment by Flamefire** [comment]
I'd say yes. It needs https://github.com/easybuilders/easybuild-easyblocks/pull/2386 though which is rotting too.

**Comment by Flamefire** [comment]
Closing, CI is in https://github.com/easybuilders/easybuild-easyconfigs/pull/24143
